futures-channel/src/lib.rs

```diff
@@ -0,0 +1,66 @@
+//! Asynchronous channels
+//!
+//! This crate provides channels which can be used to communicate between futures.
+
+#![deny(missing_docs, missing_debug_implementations)]
+#![doc(html_root_url = "https://docs.rs/futures/0.2")]
+
+#[macro_use]
+extern crate futures_core;
+
+if_std! {
+    use futures_core::{Future, Stream, Poll, Async};
+    use futures_core::task::{self, Task};
+
+    mod lock;
+    pub mod mpsc;
+    pub mod oneshot;
+}
+
+// TODO: these types should be moved to futures-sink when it stabilizes
+
+#[doc(hidden)]
+/// The result of an asynchronous attempt to send a value to a sink.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum AsyncSink<T> {
```

Comment:
Urgh this is indeed a bummer.

Could a weird combination of things like `Result<Option<T>, E>` be used instead though?

Reply:
I've switched it to `Result<Result<(), T>, E>`, but it's quite funky...